### PR TITLE
[SPARK-39000][PYTHON] Convert bools to ints in basic statistical functions of GroupBy objects

### PR DESCRIPTION
**Pull request by xinrong-meng**
### What changes were proposed in this pull request?
Convert bools to ints in basic statistical functions of GroupBy objects: `mean`, `std`, `sum`, `var`, `median`.

### Why are the changes needed?
- To reach parity with pandas.
- To be consistent with behaviors of basic statistical functions of Series/Index.

### Does this PR introduce _any_ user-facing change?
Yes.

In basic statistical functions of GroupBy objects: previously, bools were excluded; now, bools are converted to ints (following pandas).

Take `GroupBy.sum` as an example:

```py
>>> df = ps.DataFrame(
...             {
...                 "A": [1, 2, 1, 2],
...                 "B": [3.1, 4.1, 4.1, 3.1],
...                 "C": ["a", "b", "b", "a"],
...                 "D": [True, False, False, True],
...             }
...         )
>>> df
   A    B  C      D
0  1  3.1  a   True
1  2  4.1  b  False
2  1  4.1  b  False
3  2  3.1  a   True

################ PREVIOUSLY ################
>>> df.groupby("A").sum()
     B
A     
1  7.2
2  7.2

################ NOW ################
>>> df.groupby("A").sum()
     B  D
A        
1  7.2  1
2  7.2  1
```

### How was this patch tested?
Unit tests.

### CODE REVIEW

**Comment by xinrong-databricks** [comment]
@ueshin @HyukjinKwon @itholic Thank you!

**Review by HyukjinKwon — APPROVED** [review]
Looks fine.

**Review by itholic — APPROVED** [review]
Nice fix. LGTM

File: python/pyspark/pandas/tests/test_groupby.py [hunk 1]
```diff
@@ -1257,7 +1258,34 @@ def _test_stat_func(self, func):
...
-            self.assert_eq(func(p_groupby_obj).sort_index(), func(ps_groupby_obj).sort_index())
+            self.assert_eq(
+                func(p_groupby_obj).sort_index(),
+                func(ps_groupby_obj).sort_index(),
+                check_exact=check_exact,
+            )
+
+    def test_basic_stat_funcs(self):
+        self._test_stat_func(lambda groupby_obj: groupby_obj.mean(), check_exact=False)
+        self._test_stat_func(lambda groupby_obj: groupby_obj.std(), check_exact=False)
```
Comment (itholic) [inline python/pyspark/pandas/tests/test_groupby.py:1269]: Hmm... seems like the `pdf.groupby("A")[["C"]].std()` raises Exception from pandas 1.4.0 as below:

```python
>>> pdf = pd.DataFrame(
...             {
...                 "A": [1, 2, 1, 2],
...                 "B": [3.1, 4.1, 4.1, 3.1],
...                 "C": ["a", "b", "b", "a"],
...                 "D": [True, False, False, True],
...             }
...         )
>>> pdf.groupby("A")[["C"]].std()
Traceback (most recent call last):
...
ValueError: could not convert string to float: 'a'
```

Can we fix & separate the test??

Comment (xinrong-databricks) [reply python/pyspark/pandas/tests/test_groupby.py]: Good catch! May I file a follow-up PR for the fix of `std`? Now tests are separated.

**Review by HyukjinKwon — APPROVED** [review]
LGTM if tests pass. Seems there's one test failure:

```

======================================================================
ERROR [3.035s]: test_basic_stat_funcs (pyspark.pandas.tests.test_groupby.GroupByTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/__w/spark/spark/python/pyspark/testing/pandasutils.py", line 93, in assertPandasEqual
    assert_frame_equal(
  File "/usr/local/lib/python3.9/dist-packages/pandas/_testing/asserters.py", line 1308, in assert_frame_equal
    assert_series_equal(
  File "/usr/local/lib/python3.9/dist-packages/pandas/_testing/asserters.py", line 1018, in assert_series_equal
    assert_numpy_array_equal(
AssertionError: DataFrame.iloc[:, 0] (column name="B") are different

DataFrame.iloc[:, 0] (column name="B") values are different (100.0 %)
[index]: [1, 2]
[left]:  [0.7071067811865472, 0.7071067811865472]
[right]: [0.7071067811865474, 0.707106781186547]

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/__w/spark/spark/python/pyspark/pandas/tests/test_groupby.py", line 1291, in test_basic_stat_funcs
    self.assert_eq(psdf.groupby("A").std(), pdf.groupby("A").std())
  File "/__w/spark/spark/python/pyspark/testing/pandasutils.py", line 229, in assert_eq
    self.assertPandasEqual(lobj, robj, check_exact=check_exact)
  File "/__w/spark/spark/python/pyspark/testing/pandasutils.py", line 107, in assertPandasEqual
    raise AssertionError(msg) from e
AssertionError: DataFrame.iloc[:, 0] (column name="B") are different

DataFrame.iloc[:, 0] (column name="B") values are different (100.0 %)
[index]: [1, 2]
[left]:  [0.7071067811865472, 0.7071067811865472]
[right]: [0.7071067811865474, 0.707106781186547]

Left:
          B         D
A                    
B    float64
D    float64
dtype: object

Right:
          B         D
A                    
B    float64
D    float64
dtype: object

----------------------------------------------------------------------
Ran 56 tests in 465.521s
```

File: python/pyspark/pandas/tests/test_groupby.py (hunk 1 above)
Comment (itholic) [reply python/pyspark/pandas/tests/test_groupby.py]: Yeah, let's create ticket and follow it.

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**